influxql/functions.gen.go

```diff
@@ -8,12 +8,31 @@ package influxql
 
 // FloatPointAggregator aggregates points to produce a single point.
 type FloatPointAggregator interface {
-	Aggregate(p *FloatPoint)
+	AggregateFloat(p *FloatPoint)
+}
+
+// FloatBulkPointAggregator aggregates multiple points at a time.
+type FloatBulkPointAggregator interface {
+	AggregateFloatBulk(points []FloatPoint)
+}
+
+// AggregateFloatPoints feeds a slice of FloatPoint into an
+// aggregator. If the aggregator is a FloatBulkPointAggregator, it will
+// use the AggregateBulk method.
+func AggregateFloatPoints(a FloatPointAggregator, points []FloatPoint) {
```

Comment:
This function is intended to be used by Kapacitor or other programs that want to reuse code in this way. It determines which interface to use when aggregating points.

The underlying reducer may implement both the `Aggregator` and `AggregatorBulk` interfaces, but you pass in the `Aggregator` interface. It checks if it implements the `AggregatorBulk` interface and switches to using that if possible, otherwise it uses `Aggregate`.